mindmeld/converter/rasa_converter.py

```diff
@@ -0,0 +1,478 @@
+        actions = self._read_actions()
+        templates = self._read_templates()
+        # Write all functions for each action
+        self._write_functions(actions, templates, f)
+        f.close()
+        # Read contents of current file
+        file_lines = self._read_file_lines()
+        stories_dictionary = self._get_stories()
+        # Loop through all stories and create intent-action relationship
+        for item in stories_dictionary.items():
+            # Loop through steps for each story
+            for step in item[1]:
+                # Get intent, any entities, and actions
+                intent = step['intent']
+                entities = step['entities']
+                actions = step['actions']
+                # attach handle to correct function
+                app_handle_string = RasaConverter._get_app_handle(intent, entities)
+                RasaConverter._attach_handle_to_function(app_handle_string, actions[0], file_lines)
+                # check if more than 1 action per intent
+                if len(actions) > 1:
+                    RasaConverter._attach_actions_to_function(actions[1:], file_lines)
+        # write all lines back to file
+        with open(mindmeld_project_directory + "/__init__.py", "w") as f:
+            f.writelines(file_lines)
+
+    @staticmethod
+    def create_main(mindmeld_project_directory, main_file_loc):
+        with open(main_file_loc + '/rasa_main.txt', 'r') as f:
+            string = f.read()
+        with open(mindmeld_project_directory + "/__main__.py", "w") as f:
+            f.write(string)
+
+    @staticmethod
+    def create_config(mindmeld_project_directory, main_file_loc):
+        with open(main_file_loc + '/rasa_config.txt', 'r') as f:
+            string = f.read()
+        with open(mindmeld_project_directory + "/config.py", "w") as f:
+            f.write(string)
+
+    @staticmethod
+    def create_custom_features(mindmeld_project_directory, main_file_loc):
+        with open(main_file_loc + '/rasa_custom_features.txt', 'r') as f:
+            string = f.read()
+        with open(mindmeld_project_directory + "/custom_features.py", "w") as f:
+            f.write(string)
+
+    def convert_project(self):
+        """Main function that will convert a Rasa project into a Mindmeld project
+        """
+        # Create project directory with sub folders
+        self.create_mindmeld_directory(self.mindmeld_project_directory)
+        # Transfer over test data from Rasa project and reformat to Mindmeld project
+        self.create_training_data(self.rasa_project_directory, self.mindmeld_project_directory)
+        file_loc = os.path.dirname(os.path.realpath(__file__))
+        self.create_main(self.mindmeld_project_directory, file_loc)
+        self.create_init(self.mindmeld_project_directory)
+        self.create_config(self.mindmeld_project_directory, file_loc)
+        self.create_custom_features(self.mindmeld_project_directory, file_loc)
+        quit()
```

Comment:
why quit()

Reply:
I did not mean to commit this. I was testing why I saw a terminated 15 in the console in VS code.